templates/todo/projects/java-mongo/.repo/bicep/infra/main.bicep

```diff
@@ -46,154 +80,237 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
-    tags: tags
-    applicationInsightsName: monitoring.outputs.applicationInsightsName
-    appServicePlanId: appServicePlan.outputs.id
+    appInsightResourceId: appInsightResourceId
+    siteConfig: {
+      appCommandLine: './entrypoint.sh -o ./env-config.js && pm2 serve /home/site/wwwroot --no-daemon --spa'
+      linuxFxVersion: 'node|18-lts'
+      alwaysOn: true
+    }
   }
 }
 
-module webAppSettings '../../../../../../common/infra/bicep/core/host/appservice-appsettings.bicep' = {
+// Set environment variables for the frontend
+module webAppSettings 'br/public:avm/res/web/site:0.2.0' = {
   name: 'web-appsettings'
   scope: rg
   params: {
-    name: web.outputs.SERVICE_WEB_NAME
-    appSettings: {
-      REACT_APP_API_BASE_URL: useAPIM ? apimApi.outputs.SERVICE_API_URI : api.outputs.SERVICE_API_URI
-      REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING: monitoring.outputs.applicationInsightsConnectionString
+    kind: 'app'
+    name: web.outputs.name
+    serverFarmResourceId: appServicePlan.outputs.resourceId
+    tags: union(tags, { 'azd-service-name': 'web' })
+    appSettingsKeyValuePairs: {
+      REACT_APP_API_BASE_URL: 'https://${api.outputs.defaultHostname}'
+      REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING: applicationInsights.outputs.connectionString
     }
   }
 }
 
 // The application backend
-module api '../../../../../common/infra/bicep/app/api-appservice-java.bicep' = {
+module api 'br/public:avm/res/web/site:0.2.0' = {
   name: 'api'
   scope: rg
   params: {
+    kind: 'app'
     name: !empty(apiServiceName) ? apiServiceName : '${abbrs.webSitesAppService}api-${resourceToken}'
+    serverFarmResourceId: appServicePlan.outputs.resourceId
+    tags: union(tags, { 'azd-service-name': 'api' })
     location: location
-    tags: tags
-    applicationInsightsName: monitoring.outputs.applicationInsightsName
-    appServicePlanId: appServicePlan.outputs.id
-    keyVaultName: keyVault.outputs.name
-    allowedOrigins: [ web.outputs.SERVICE_WEB_URI ]
-    appSettings: {
-      AZURE_COSMOS_CONNECTION_STRING_KEY: cosmos.outputs.connectionStringKey
-      AZURE_COSMOS_DATABASE_NAME: cosmos.outputs.databaseName
-      AZURE_COSMOS_ENDPOINT: cosmos.outputs.endpoint
-      API_ALLOW_ORIGINS: web.outputs.SERVICE_WEB_URI
+    appInsightResourceId: appInsightResourceId
+    managedIdentities: {
+      systemAssigned: true
+    }
+    siteConfig: {
+      cors: {
+        allowedOrigins: [ 'https://portal.azure.com', 'https://ms.portal.azure.com' ,'https://${web.outputs.defaultHostname}' ]
+      }
+      linuxFxVersion: 'java|17-java17'
+      alwaysOn: true
+      appCommandLine: ''
+    }
+    appSettingsKeyValuePairs: {
+      AZURE_KEY_VAULT_ENDPOINT: keyVault.outputs.uri
+      AZURE_COSMOS_CONNECTION_STRING_KEY: connectionStringKey
+      AZURE_COSMOS_DATABASE_NAME: !empty(cosmosDatabaseName) ? cosmosDatabaseName: 'Todo'
+      AZURE_COSMOS_ENDPOINT: 'https://${cosmos.outputs.name}.mongo.cosmos.azure.com:443/'
```

Comment:
I'm wondering if that fqdn can be derived from an output instead of hardcoded?  Does the cosmos service provide that as an output?

Reply:
The cosmos module in AVM does not provide the output `endpoint`.
Link: https://github.com/Azure/bicep-registry-modules/tree/main/avm/res/document-db/database-account#Outputs

Reply:
Can you confirm that we have an issue to track that and link it here?